controller/api/models.py

```
@@ -1129,10 +1131,12 @@ def _etcd_purge_user(**kwargs):
         pass
 
 
-def _etcd_create_app(**kwargs):
+def _etcd_publish_app(**kwargs):
     appname = kwargs['instance']
-    if kwargs['created']:
+    try:
         _etcd_client.write('/deis/services/{}'.format(appname), None, dir=True)
+    except KeyError:
```

This could be just `# Ignore error when the directory already exists.`